lib/tests/modules.sh

```diff
@@ -13,126 +16,121 @@ fail=0
 
 evalConfig() {
     local attr=$1
-    shift;
-    local script="import ./default.nix { modules = [ $@ ];}"
+    shift
+    local script="import ./default.nix { modules = [ $* ];}"
     nix-instantiate --timeout 1 -E "$script" -A "$attr" --eval-only --show-trace --read-write-mode
 }
 
 reportFailure() {
     local attr=$1
-    shift;
-    local script="import ./default.nix { modules = [ $@ ];}"
+    shift
+    local script="import ./default.nix { modules = [ $* ];}"
     echo 2>&1 "$ nix-instantiate -E '$script' -A '$attr' --eval-only"
-    evalConfig "$attr" "$@"
-    fail=$((fail + 1))
+    evalConfig "$attr" "$@" || true
+    ((++fail))
 }
 
 checkConfigOutput() {
     local outputContains=$1
-    shift;
+    shift
     if evalConfig "$@" 2>/dev/null | grep --silent "$outputContains" ; then
-        pass=$((pass + 1))
-        return 0;
+        ((++pass))
     else
         echo 2>&1 "error: Expected result matching '$outputContains', while evaluating"
         reportFailure "$@"
-        return 1
     fi
 }
 
 checkConfigError() {
     local errorContains=$1
     local err=""
-    shift;
-    if err==$(evalConfig "$@" 2>&1 >/dev/null); then
+    shift
+    if err="$(evalConfig "$@" 2>&1 >/dev/null)"; then
         echo 2>&1 "error: Expected error code, got exit code 0, while evaluating"
         reportFailure "$@"
-        return 1
     else
         if echo "$err" | grep -zP --silent "$errorContains" ; then
-            pass=$((pass + 1))
-            return 0;
+            ((++pass))
```

Comment:
Do you remember under which number shellcheck suggests that?

Reply:
ShellCheck doesn't detect any of these changes. I removed the semicolon because it's unnecessary noise, I removed the `return` because the return code isn't used, and I changed from `pass=$((pass + 1))` to `((++pass))` because I've had to do that elsewhere to ensure that I don't get a zero arithmetic result, which results in a non-zero exit code.